lusee/simulation.py

```diff
@@ -29,18 +29,37 @@ class Simulator:
 
     def __init__ (self, obs, beams, sky_model, 
                   combinations = [(0,0),(1,1),(0,2),(1,3),(1,2)], lmax = 128,
-                  taper = 0.03, Tground = 200.0, freq_ndx=None):
+                  taper = 0.03, Tground = 200.0, freq = None):
         self.obs = obs
         self.sky_model = sky_model
         self.lmax = lmax
         self.taper = taper
         self.Tground = Tground
-        if freq_ndx is None:
-            self.freq_ndx = np.arange(beams[0].Nfreq)
+        if freq is None:
+            self.freq = beams[0].freq
         else:
-            self.freq_ndx = freq_ndx
-        self.freq=beams[0].freq[self.freq_ndx]
-
+            self.freq = freq
+            
+        freq_ndx_beam = []
+        freq_ndx_sky = []
+        for f in self.freq:
+            try:
+                ndx = list(beams[0].freq).index(f)
+            except ValueError:
+                print ("Error:")
+                print (f"Frequency {f} does not exist in beams.")
+                sys.exit(1)
+            freq_ndx_beam.append(ndx)
+            try:
+                ndx = list(sky_model.freq).index(f)
+            except ValueError:
+                print ("Error:")
+                print (f"Frequency {f} does not exist in sky model.")
+                sys.exit(1)
```

Comment:
This needs a `import sys` (although I guess erroring will also exit...)

Reply:
well spotted! :)